### PR TITLE
Adding x-ms-capabilities

### PR DESCRIPTION
**Pull request by sarunassp**
Adding file-picker support

### CODE REVIEW

File: src/SwashBuckle.AspNetCore.MicrosoftExtensions/VendorExtensionEntities/FilePickerOperationModel.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        /// <summary>
+        /// Lookup operation ID, use swagger operation ID of action to call
+        /// </summary>
+        [JsonProperty("operation-id")]
+        public string OperationId { get; }
+
+        /// <summary>
+        /// Parameter value to pass to lookup operation
+        /// (e.g., lookupOpParam={paramNameFromThisOperation}&amp;lookupOpParam2=hardcoded)
```
Comment (BenasJacikas) [inline src/SwashBuckle.AspNetCore.MicrosoftExtensions/VendorExtensionEntities/FilePickerOperationModel.cs]: Remove example

File: src/SwashBuckle.AspNetCore.MicrosoftExtensions/Helpers/ParameterParser.cs [hunk 2]
```diff
@@ -25,7 +25,18 @@ internal class ParameterParser
                     new Dictionary<string, string> {{Constants.Parameter, matches.Groups[1].Value}}
                 );
             }
-            return new KeyValuePair<string, object>(parameter.Key, parameter.Value[0]);
+
+            switch (parameter.Value[0])
```
[the remark this answers is not in the document]
Comment (sarunassp) [reply src/SwashBuckle.AspNetCore.MicrosoftExtensions/Helpers/ParameterParser.cs]: true/false can be used as parameters so they need to be bool and not strings
snippet from google's swagger

                                        "name": "path",
 					"in": "query",
 					"description": "Path of the file in Google Drive",
 					"required": true,
 					"x-ms-dynamic-values": {
 						"capability": "file-picker",
 						"parameters": {
 							"isFolder": false,
 							"fileFilter": [],
 							"dataset": null
 						},
 						"value-path": "Path"
 					},
 					"x-ms-summary": "File path",
 					"type": "string"

File: src/SwashBuckle.AspNetCore.MicrosoftExtensions/VendorExtensionEntities/FilePickerOperationModel.cs [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        /// </param>
+        public FilePickerOperationModel (string operationId, Dictionary<string, string> parameters)
+        {
+            //no operation - dont create parameters
+            if (string.IsNullOrEmpty(operationId))
+                return;
+
+            OperationId = operationId;
+            if (parameters == null)
```
[the remark this answers is not in the document]
Comment (sarunassp) [reply src/SwashBuckle.AspNetCore.MicrosoftExtensions/VendorExtensionEntities/FilePickerOperationModel.cs]: They wouldn't be generated in swagger

File: src/SwashBuckle.AspNetCore.MicrosoftExtensions/Helpers/ParameterParser.cs [hunk 4]
```diff
@@ -25,7 +25,19 @@ internal class ParameterParser
...
                 );
             }
-            return new KeyValuePair<string, object>(parameter.Key, parameter.Value[0]);
+
+            // parse true/false values as bools
+            switch (parameter.Value[0])
+            {
+                case "true":
+                    return new KeyValuePair<string, object>(parameter.Key, true);
+                    break;
```
Comment (BenasJacikas) [inline src/SwashBuckle.AspNetCore.MicrosoftExtensions/Helpers/ParameterParser.cs]: No need for break when return is present

File: src/SwashBuckle.AspNetCore.MicrosoftExtensions/Helpers/ParameterParser.cs [hunk 5]
```diff
@@ -25,7 +25,19 @@ internal class ParameterParser
...
+                case "false":
+                    return new KeyValuePair<string, object>(parameter.Key, false);
+                    break;
```
Comment (BenasJacikas) [inline src/SwashBuckle.AspNetCore.MicrosoftExtensions/Helpers/ParameterParser.cs]: No need for break

**Pull request merged**